Добавляет разметку и стили шрифтов

RisenPhoenixSpB opened the pull request.

RisenPhoenixSpB committed:
Добавляет хедер и часть декоративных элементов

RisenPhoenixSpB committed:
Дорабатывает хедер, добавляет часть main

RisenPhoenixSpB committed:
Добавляет верстку index.html

RisenPhoenixSpB committed:
Дорабатывает index.html

RisenPhoenixSpB committed:
Добавляет разметку index.html, catalog.html

RisenPhoenixSpB committed:
Добавляет правки и добавляет модальное окно

RisenPhoenixSpB committed:
Добавляет стили шрифтов

AlSudar commented on index.html:
```diff
@@ -1,12 +1,547 @@
+          <img class="device-logo-image" src=images/device-logo.svg alt="Логотип Девайса">
+        </a>
+        <div class="user-features">
+          <!--Желтый фон будет сделан ::before или ::after-->
```
Почему именно так?

AlSudar commented on index.html:
```diff
@@ -1,12 +1,547 @@
+              <button class="form-button" type="submit" name="search">Найти</button>
+            </form>
+          </div>
+          <div class="user-account-container">
```
Мне кажется этот контейнер лишний

AlSudar commented on index.html:
```diff
@@ -1,12 +1,547 @@
+              Войти
+            </a>
+          </div>
+          <div class="comparison-container">
```
Мне кажется этот контейнер лишний

AlSudar commented on index.html:
```diff
@@ -1,12 +1,547 @@
+                  замаскированным под стандартную камеру.</p>
+                <a class="slide-detailed" href="#">Подробнее</a>
+                <div class="slide-specifications">
+                  <h3 class="flight-range-value">800 м</h3>
```
Мне кажется здесь уместнее использовать теги dl/dt, т.к. здесь идёт речь о технических характеристиках

AlSudar commented on index.html:
```diff
@@ -1,12 +1,547 @@
+      <section class="product-categories">
+        <h2 class="visually-hidden">Категории товаров</h2>
+        <ul class="product-categories-list">
+          <li class="product-categories-card">
```
Эти элементы - это ссылки на другие страницы, это можно понять исходя из того, что в стайлгайде при наведении на них меняется их фоновый цвет

AlSudar commented on index.html:
```diff
@@ -1,12 +1,547 @@
+        <h2 class="visually-hidden">Дополнительные услуги</h2>
+        <div class="service-tabs">
+          <div class="service-tabs-buttons">
+            <button class="service-tab-button service-tab-current" type="button">Доставка</button>
```
Мне кажется здесь можно использовать тег для списка, это же список кнопок

AlSudar commented on index.html:
```diff
@@ -1,12 +1,547 @@
+        </div>
+        <h2 class="rare-delivery-title">Доставим редкий товар под заказ</h2>
+        <div class="rare-delivery-information-container">
+          <span class="circle-information">i</span>
```
Это должно быть иконкой а не буквой

AlSudar commented on index.html:
```diff
@@ -1,12 +1,547 @@
+        <img class="footer-device-logo" src="images/footer/device-logo-footer.svg" alt="Логотип Девайса" width="145"
+          height="36">
+        <address class="footer-address">
+          <p class="footer-address-text">Санкт-Петербург, набережная реки Карповки, 5, литера П.</p>
```
Мне кажется что текст можно обернуть сразу в address, без параграфа

AlSudar reviewed (commented):
У тебя получилась хорошая разметка, особенно мне понравился большой уровень доступности. Однако нашел несколько мест где можно сделать лучше, обработай мои комментарии, и после этого я померджу

RisenPhoenixSpB committed:
Добавляет правки

Также добавил пример своего видения как реализовать фоновый желтый цвет у хедера (полностью размечать не стал) - ответ на самый первый комментарий.

RisenPhoenixSpB committed:
Добавляет полную стилизацию header внутри index

Включает в себя pixel perfect. Также добавляет небольшие правки в разметку.

RisenPhoenixSpB committed:
Добавляет стилизацию слайдера (1 слайд)

RisenPhoenixSpB committed:
Добавляет стили index.html и модальному окну

Пока не добавил стилизацию поповера у корзины

AlSudar commented on catalog.html:
```diff
@@ -0,0 +1,433 @@
+            <a class="breadcrumb-goods-catalog" href="#">Каталог товаров</a>
+          </div>
+          <div class="breadcrumb-decorative">
+            <svg class="breadcrumb-arrow" aria-hidden="true" focusable="false" xmlns="http://www.w3.org/2000/svg"
```
Стрелочку нужно добавить через псевдо-элемент

AlSudar commented on catalog.html:
```diff
@@ -0,0 +1,433 @@
+        <section class="filters-and-sorting">
+          <div class="filters-container">
+            <h2 class="filters-title">Фильтр:</h2>
+            <form class="catalog-filters" action="https://echo.htmlacademy.ru/" method="get">
```
Это не особо важно, но лучше уточню:
Метод get у форм служит для получения информации с сервера
А метод post для отправления информации на сервер. В данном случае ты отправляешь информацию на сервер

AlSudar commented on catalog.html:
```diff
@@ -0,0 +1,433 @@
+              <option class="sorting-product-item" value="expensive">Дороже</option>
+            </select>
+            <div class="sorting-buttons">
+              <span class="sorting-button-up"></span>
```
Логичнее сделать это кнопками

AlSudar commented on catalog.html:
```diff
@@ -0,0 +1,433 @@
+            <ul class="pagination-list">
+              <li class="pagination-item">
+                <a class="pagination-button" href="#">
+                  <span class="pagination-number pagination-number-current">1</span>
```
span - это лишняя обёртка в данном случае. И логичнее здесь использовать button а не тег ссылки, ты же не переходишь на другую страницу

AlSudar commented on catalog.html:
```diff
@@ -0,0 +1,433 @@
+          <ul class="footer-social-list">
+            <li class="footer-social-item">
+              <a class="footer-social-link" href="https://t.me/htmlacademy">
+                <span class="social-circle-element">
```
Мне кажется это лишняя обёртка

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+  display: flex;
```
Необязательно здесь использовать flex, блоки внутри body и так расположатся друг под другом
А вот дефолтный font-size и line-heigth лучше задать

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+}
+
+.navigation {
+  position: absolute;
```
Зачем абсолютно позиционировать этот блок? Лучше сделать relative, тогда он останется в поток документа, и можно будет пользоваться right и top

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+  position: absolute;
+  right: 60px;
+  top: -20px;
+  display: flex;
```
Здесь можно обойтись без flex, по той же причине что и в body

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+}
+
+.device-logo-link {
+  display: flex;
```
Зачем здесь flex?

RisenPhoenixSpB replied:
Если убрать флекс, то ссылка будет немного больше, чем само изображение по высоте. Чтобы выровнять это, использовал flex.

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+}
+
+
+.searching-form-image {
```
Мне кажется эту картинку лучше добавить через псевдоэлемент

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+  min-height: 120px;
+}
+
+.good-catalog-popover-column-third .good-catalog-popover-column-item {
```
Мне кажется что одним пикселем можно и погрешить

RisenPhoenixSpB replied:
Не совсем понял мысль. Я задал min-height на случай, если будут добавляться дополнительные пункты меню.

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+  height: 560px;
+}
+
+/*Это класс будет переноситься при переключении слайдов, чтобы показывать нужный */
```
Вам не нужно реализовывать переключение слайдов

RisenPhoenixSpB replied:
Решил так сделать исходя из логики, которая транслирует html академия. Чтобы разработчик js, которому я передам текущую верстку, понимал, что слайды переключаются именно этим классом.

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+  color: #444444;
+}
+
+/*Почему, если здесь убрать у ссылки и или before z-index (или одновременно у обоих), то желтая полоса не появится?*/
```
Не очень понял сути вопроса

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+  margin: 0;
+}
+
+.stick-length {
```
Зачем создавать отдельные стили для этого? Можно же использовать те же что и у первого слайда, нет?

AlSudar commented on styles/styles.css:
```diff
@@ -0,0 +1,2150 @@
+}
+
+.form-icon-mail {
+  position: absolute;
```
Мне кажется можно создать единый класс для иконок формы, и использовать его. А для подключения картинки к иконки добавлять второй класс, уникальный для каждой

AlSudar reviewed (commented):
Разметка получилась хорошей, однако у тебя много дублирования кода от которого стоит избавиться

RisenPhoenixSpB committed:
Исправляет ошибки в index.html, добавляет стили catalog.html

Поработал над комментариями в index.html (разметка + стили) и добавил стили catalog.html. Всё это пока без эффектов при наведении и т.д.

RisenPhoenixSpB committed:
Правит стили пагинации catalog.html
